### DISCUSSION TITLE
Does Ryven have potential to help me solve my problem?

### DISCUSSION THREAD

**Discussion by paul-s-johnson** [Tanneguydv/Pythonocc-nodes-for-Ryven#15] [Q&A]
I am in the early stages of trying to find a solution to a work problem. We have a lot of semi-complicated interest calculations to make with enough variation in them that they don't lend themselves well to storing them in regular rows and columns (i.e. a relational database). It also involves non-programmers to maintain them. So my initial thought is to have GUI for technically-oriented, non-programmers to manage the calculations. Custom nodes could be made by the programmers, e.g. to fetch a value from a remote source, etc. 

I've installed and played around with Ryven and it seems pretty polished for something at this stage of development. But the documentation so far is scant and some things are not very clear.

What I want to ask is if Ryven has the potential to be a part of my solution. For example, can a Ryven script be triggered remotely in Python handing it input and retrieving its output? I would want to use Ryven to manage calculations, but then trigger the input and output outside of Ryven in batch jobs. 

At any rate, Ryven seems pretty cool with a lot of potential, but I'm still trying to figure out what that is, so thank you for your patience and for making this available.

Paul

**Comment by Tanneguydv** [comment]
Hello and welcome,
I feel that your question is more around Ryven than Pythonocc nodes for Ryven? Or do you use Pythonocc in your work?
I can't answer completely  to your question, I think @leon-thomm could help and that [ryvencore](https://github.com/leon-thomm/ryvencore) is part of the answer

**Comment by leon-thomm** [comment]
Thanks for the ping! To your question:

If it's only going to need to fit those constraints of remote deployment, then **yes**, Ryven can be the answer to this, or more precisely, as Tanneguy already suspected, [ryvencore](https://github.com/leon-thomm/ryvencore) makes it possible. While Ryven has ryvencore deeply integrated (so Ryven does in particular *not* serve as a UI to a backend/ryvencore session running remotely), you can build and test something locally in Ryven, then export the project file and deploy it manually on ryvencore. And once you have a fixed setup, it should be easy to completely automize this workflow. 

Also, you cannot change a project while it's running simply on ryvencore (well, you actually could write a node that sort of does this by modifying its own graph, but let's not go down this rabbit hole), if you want to make changes you need to do them in Ryven and export and deploy a new project file. But if that's not a problem, then what you described so far sounds doable, at least automatic remote deployment, automatic tests, etc. should be easy to realize, especially since Ryven became a Python package.

That being said, Ryven only provides the UI and nodes execution engine (ryvencore), but it doesn't really provide any actual node functionality. To provide a framework of nodes that is intuitive and easy to use and which works reliably, it's still going to require quite some effort to make these nodes.

All the best
Leon
